routing/routing.go

```diff
@@ -361,6 +377,12 @@ func (r *Routing) startReceivingUpdates(o Options) {
 		for {
 			select {
 			case rt := <-c:
+				var rtToClose *routeTable
+				rtToClose, ok := r.routeTable.Load().(*routeTable)
+				if ok {
+					rtToClose.close()
```

Comment:
pretty sure this leads to a lot of 404. If you need to close it then after the Store below.